### PR TITLE
Adds version field while listing all the tasks

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
- This patch list the task with the version of the task

   For e.g.

       ./tkn task list
       NAME                DESCRIPTION              VERSION   AGE
       ansible-tower-cli   Ansible-tower-cli t...   0.1       21 hours ago
       buildah                                      ---       20 hours ago 
      
Partially closes: #1264 

Signed-off-by: Puneet Punamiya <ppunamiy@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

### CODE REVIEW

File: pkg/cmd/task/list.go [hunk 1]
```diff
@@ -36,18 +36,18 @@ No Tasks found
...
+{{ end -}}
 {{- end -}}
 {{- range $_, $t := .Tasks.Items }}{{- if $t }}
 {{- if $.AllNamespaces -}}
-{{ $t.Namespace }}	{{ $t.Name }}	{{ formatDesc $t.Spec.Description }}	{{ formatAge $t.CreationTimestamp $.Time }}
-{{ else -}} 
-{{ $t.Name }}	{{ formatDesc $t.Spec.Description }}	{{ formatAge $t.CreationTimestamp $.Time }}
+{{ $t.Namespace }}	{{ $t.Name }}	{{ formatDesc $t.Spec.Description }}	{{ formatVersion $t.Labels }}	{{ formatAge $t.CreationTimestamp $.Time }}
+{{ else -}}
+{{ $t.Name }}	{{ formatDesc $t.Spec.Description }}	{{ formatVersion $t.Labels }}	{{ formatAge $t.CreationTimestamp $.Time }}
```
Comment (vinamra28) [inline pkg/cmd/task/list.go:48]: can we also add a test in which a task has a version?

File: pkg/formatted/version.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package formatted
+
+func FormatVersion(t map[string]string) string {
```
Comment (vinamra28) [inline pkg/formatted/version.go:17]: can we have a better variable instead of `t`?

File: test/builder/builder.go [hunk 3]
```diff
@@ -295,8 +295,8 @@ func ListAllTasksOutput(t *testing.T, cs *framework.Clients, td map[int]interfac
 	t.Helper()
 	const (
 		emptyMsg = "No tasks found"
-		header   = "NAME\tDESCRIPTION\tAGE"
-		body     = "%s\t%s\t%s\n"
+		header   = "NAME\tDESCRIPTION\tVERSION\tAGE"
+		body     = "%s\t%s\t%s\t%s\n"
```
Comment (vinamra28) [inline test/builder/builder.go:299]: can we update the e2e tests also to have a version?

**Comment by chmouel** [comment]
Most tasks that are not coming from hub (which i assume is most of the tasks user wrote) would not have the version field which would then always shows "---"

I am not sure I am too keen on this "---" showing all the time, is there a better way ?

**Comment by PuneetPunamiya** [comment]
@chmouel  Thank you for the review, would it be fine if we show nothing instead of "---" as for the description as well we don't show anything if there is no description

**Comment by chmouel** [comment]
i guess my issue is that the version label is not a top level API field while description it is i.e: users would hardly would know that we can have task versioning it's just a convention kind of thing.

So in practice only the tasks from the official catalog are going to have it. So for most people this is not going to be any useful and only show empty....

**Comment by PuneetPunamiya** [comment]
Creating a different pr to add version field in describe command and not in list command
/close

**Pull request closed without merging**